classes/core/APIRouter.inc.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * @file classes/core/APIRouter.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class APIRouter
+ * @ingroup core
+ *
+ * @brief Map HTTP requests to a REST API using the Slim microframework.
+ *
+ * Requests for [index.php]/api are intercepted for site-level API requests,
+ * and requests for [index.php]/{contextPath}/api are intercepted for
+ * context-level API requests.
+ */
+
+import('lib.pkp.classes.core.PKPRouter');
+import('classes.core.Request');
+import('classes.handler.Handler');
+
+class APIRouter extends PKPRouter {
+	/** @var APIHandler */
+	var $_handler;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		parent::__construct();
+	}
+
+	/**
+	 * Determines whether this router can route the given request.
+	 * @param $request PKPRequest
+	 * @return boolean true, if the router supports this request, otherwise false
+	 */
+	function supports($request) {
+		if (!isset($_SERVER['PATH_INFO'])) return false;
+		$pathInfoParts = explode('/', trim($_SERVER['PATH_INFO'], '/'));
+
+		if (count($pathInfoParts)>=2 && $pathInfoParts[1] == 'api') {
+			// Context-specific API requests: [index.php]/{contextPath}/api
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get the API version
+	 * @return string
+	 */
+	function getVersion() {
+		$pathInfoParts = explode('/', trim($_SERVER['PATH_INFO'], '/'));
+		return Core::cleanFileVar(isset($pathInfoParts[2]) ? $pathInfoParts[2] : '');
+	}
+
+	/**
+	 * Get the entity being requested
+	 * @return string
+	 */
+	function getEntity() {
+		$pathInfoParts = explode('/', trim($_SERVER['PATH_INFO'], '/'));
```

Comment:
Uff. Ok, I will investigate.

Maybe use `context` instead of `journal`? I wonder if we'll have any name collisions with endpoint arguments. Maybe we can prefix the path info params?